packages/salesforcedx-vscode-core/test/vscode-integration/commands/templates/forceApexClassCreate.test.ts

```diff
@@ -7,31 +7,122 @@
+        fileName,
+        outputdir: outputDirPath
+      });
+      expect(classCreateCommand.toCommand()).to.equal(
+        `sfdx force:apex:class:create --classname ${fileName} --template DefaultApexClass --outputdir ${outputDirPath}`
+      );
+      expect(classCreateCommand.description).to.equal(
+        nls.localize('force_apex_class_create_text')
+      );
+      expect(classCreate.getDefaultDirectory()).to.equal('classes');
+      expect(classCreate.getFileExtension()).to.equal('.cls');
+      expect(
+        classCreate
+          .getSourcePathStrategy()
+          .getPathToSource(outputDirPath, fileName, '.cls')
+      ).to.equal(path.join(outputDirPath, `${fileName}.cls`));
+    });
+  });
+
+  describe('Library Create', () => {
+    let settings: SinonStub;
+    let showInputBoxStub: SinonStub;
+    let quickPickStub: SinonStub;
+    let appendLineStub: SinonStub;
+    let showSuccessfulExecutionStub: SinonStub;
+    let showFailedExecutionStub: SinonStub;
+
+    beforeEach(() => {
+      // mock experimental setting
+      settings = stub(SfdxCoreSettings.prototype, 'getTemplatesLibrary');
+      settings.returns(true);
+      showInputBoxStub = stub(vscode.window, 'showInputBox');
+      quickPickStub = stub(vscode.window, 'showQuickPick');
+      appendLineStub = stub(channelService, 'appendLine');
+      showSuccessfulExecutionStub = stub(
+        notificationService,
+        'showSuccessfulExecution'
+      );
+      showSuccessfulExecutionStub.returns(Promise.resolve());
+      showFailedExecutionStub = stub(
+        notificationService,
+        'showFailedExecution'
+      );
+    });
+
+    afterEach(() => {
+      settings.restore();
+      showInputBoxStub.restore();
+      quickPickStub.restore();
+      showSuccessfulExecutionStub.restore();
+      showFailedExecutionStub.restore();
+      appendLineStub.restore();
+    });
+
+    it('Should create Apex Class', async () => {
+      // arrange
+      showInputBoxStub.returns('TestApexClass');
+      quickPickStub.returns('force-app/main/default/classes');
+      const apexClassPath = path.join(
+        getRootWorkspacePath(),
+        'force-app/main/default/classes',
+        'TestApexClass.cls'
+      );
+      const apexClassMetaPath = path.join(
+        getRootWorkspacePath(),
+        'force-app/main/default/classes',
+        'TestApexClass.cls-meta.xml'
+      );
+      shell.rm('-f', apexClassPath);
+      shell.rm('-f', apexClassMetaPath);
+      assert.noFile([apexClassPath, apexClassMetaPath]);
+
+      // act
+      await forceApexClassCreate();
+
+      // assert
+      assert.file([apexClassPath, apexClassMetaPath]);
+      assert.fileContent(
+        apexClassPath,
+        'public with sharing class TestApexClass'
+      );
+      assert.fileContent(
+        apexClassMetaPath,
+        `<?xml version="1.0" encoding="UTF-8"?>
+<ApexClass xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>49.0</apiVersion>
```

Comment:
We should also expose the default API version in library API, so that we don't have to update every release in tests.